packages/strapi-plugin-content-type-builder/services/schema-builder/content-type-builder.js

```diff
@@ -174,6 +177,8 @@ module.exports = function createComponentBuilder() {
             this.unsetRelation(oldAttribute);
           }
 
+          newAttribute.autoPopulate = oldAttribute.autoPopulate;
```

Comment:
I just tested your branch and I discovered an issue with this statement :)
I tried not setting the autoPopulate first and then call the api with an autopopulate value. This will override the edit and leave autopopulate to undefined. (Really a edge case because the interface doesn't offer the option)

For the sake of completness could you please check do sth like

`newAttribute.autoPopulate = newAttribute.autoPopulate || oldAttribute.autoPopulate` ? 

Thanks !